src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -151,20 +157,34 @@ int BasePlaylistFeature::playlistIdFromIndex(const QModelIndex& index) {
     }
 }
 
+void BasePlaylistFeature::selectPlaylistInSidebar(int playlistId) {
+    if (playlistId == -1) {
+        return;
+    }
+    QModelIndex index = indexFromPlaylistId(playlistId);
+    if (index.isValid() && m_pSidebarWidget) {
+        m_pSidebarWidget->selectChildIndex(index);
+    }
+}
+
 void BasePlaylistFeature::activateChild(const QModelIndex& index) {
     //qDebug() << "BasePlaylistFeature::activateChild()" << index;
     int playlistId = playlistIdFromIndex(index);
     if (playlistId != -1) {
         m_pPlaylistTableModel->setTableModel(playlistId);
         emit showTrackModel(m_pPlaylistTableModel);
         emit enableCoverArtDisplay(true);
+        if (m_pSidebarWidget) {
+            m_pSidebarWidget->selectChildIndex(index);
+        }
     }
 }
 
 void BasePlaylistFeature::activatePlaylist(int playlistId) {
-    //qDebug() << "BasePlaylistFeature::activatePlaylist()" << playlistId;
+    // qDebug() << "BasePlaylistFeature::activatePlaylist()" << playlistId;
     QModelIndex index = indexFromPlaylistId(playlistId);
     if (playlistId != -1 && index.isValid()) {
```

Comment:
One more, early return if playlistId == -1. I will not add more comments about this issue, please check all new or edited functions.